lib.php

```diff
@@ -177,8 +177,11 @@ public function get_links($linkarray) {
         if (!empty($linkarray["file"])) {
             $file = $linkarray["file"];
             $filearea = $file->get_filearea();
+
             $nonsubmittingareas = array("feedback_files", "introattachment");
-            if (in_array($filearea, $nonsubmittingareas)) {
+            $allowedcomponents = array("assignsubmission_file", "mod_assign", "mod_forum", "mod_workshop", "question");
```

Comment:
Are quiz submissions covered by one of these component names?

Reply:
That's what "question" is. I imagine there may be multiple quiz component types, but as we only support essay questions for plagiarism it's not something to be concerned about.